crates/wasmtime/src/component/func.rs

```diff
@@ -37,15 +37,20 @@ use wasmtime_runtime::{Export, ExportFunction, VMTrampoline};
 #[doc(hidden)]
 #[macro_export]
 macro_rules! map_maybe_uninit {
-    ($maybe_uninit:ident $($field:tt)*) => (#[allow(unused_unsafe)] unsafe {
-        use $crate::component::__internal::MaybeUninitExt;
-
-        let m: &mut std::mem::MaybeUninit<_> = $maybe_uninit;
-        // Note the usage of `addr_of_mut!` here which is an attempt to "stay
-        // safe" here where we never accidentally create `&mut T` where `T` is
-        // actually uninitialized, hopefully appeasing the Rust unsafe
-        // guidelines gods.
-        m.map(|p| std::ptr::addr_of_mut!((*p)$($field)*))
+    ($maybe_uninit:ident $($field:tt)*) => ({
+        #[allow(unused_unsafe)]
+        {
+            unsafe {
+                use $crate::component::__internal::MaybeUninitExt;
+
+                let m: &mut std::mem::MaybeUninit<_> = $maybe_uninit;
+                // Note the usage of `addr_of_mut!` here which is an attempt to "stay
+                // safe" here where we never accidentally create `&mut T` where `T` is
+                // actually uninitialized, hopefully appeasing the Rust unsafe
+                // guidelines gods.
+                m.map(|p| std::ptr::addr_of_mut!((*p)$($field)*))
+            }
+        }
     })
 }
```

Comment:
This change is not related to this PR on the surface, but it fixes a compile error on nightly Rust which will be required to get fuzzing back online.

Reply:
(and it's also not useful to land this independently because we also need to update `wasm-mutate` which needs an update to `wasmparser` which needs ...)